### PR TITLE
Add docs for platform apps support with editor flow

### PR DESCRIPTION
**Pull request by yavorsky**
### 🔦 Summary

Screen Shot 2020-04-21 at 1 06 17 PM

Since we've integrated platform apps support to the editor flow, we need to keep in sync documentation related to the changes.
This is a first stage of the docs.

We are not going to serve it like this in future and we'll add some pretty cool hipster website.
But for now it's pretty ok to maintain some raw version located in the editor flow repo to share with new potential users.

This is just basic concept and we will add more sections / extend current ones in future.

### CODE REVIEW

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+### Intro
+
+<img width="790" alt="Screen Shot 2020-04-21 at 1 06 17 PM" src="https://user-images.githubusercontent.com/1521229/79906725-028fff00-8421-11ea-934b-660adef18180.png">
+
+After understanding that we have a lot of things in common between platform apps and out-of-iframe apps,
+we decided to combine efforts and allow yoshi to support platform apps.
+This means 2 big things for the future of the development apps inside Wix:
+
+Firstly, if you are developing out-of-iframe app and want to migrate it gradually to platform app, you don't have to create a new project. Imagine you have 2 components in your out-of-iframe app. It's product's list and product's page.
+You want to allow your users to take all from thebenefits platform's apps. Configuring each element in the list using already created native components sounds promising!
```
[the remark this answers is not in the document]
Comment (yavorsky) [reply packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:14]: thanks.. weird.. double checked with grammarly and still missed some types :(

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+### Intro
+
+<img width="790" alt="Screen Shot 2020-04-21 at 1 06 17 PM" src="https://user-images.githubusercontent.com/1521229/79906725-028fff00-8421-11ea-934b-660adef18180.png">
+
+After understanding that we have a lot of things in common between platform apps and out-of-iframe apps,
+we decided to combine efforts and allow yoshi to support platform apps.
+This means 2 big things for the future of the development apps inside Wix:
+
+Firstly, if you are developing out-of-iframe app and want to migrate it gradually to platform app, you don't have to create a new project. Imagine you have 2 components in your out-of-iframe app. It's product's list and product's page.
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:13]:
```suggestion
Firstly, if you are developing out-of-iframe app and want to migrate it gradually to platform app, you don't have to create a new project. Imagine you have 2 components in your out-of-iframe app: products list and product page.
```

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+You want to allow your users to take all from thebenefits platform's apps. Configuring each element in the list using already created native components sounds promising!
+But creating a new project from scratch could take time. A lot of functional logic will be just copy-pasted. You basically have to remove `Widget.ts` containing
+React Component and make all the UI magic inside the viewer controller. So by using editor flow you can just do this w/o new bootstrap.
+Yoshi is smart enough to understand what kind of component it is to handle.
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:17]:
```suggestion
Yoshi is smart enough to understand what kind of component it needs to handle.
```

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+The second win is a learning curve. We are trying to find more things in common between the platform and OOI apps, so most of the configuration options, way we are developing
+locally, serving Settings Panel App, running tests, etc should be almost the same. Both types of apps have a controller for viewer part -`viewer.controller.ts`, entry points for worker bundles: `viewer.app.ts`, `editor.app.ts`, rendered via React Settings panel - `Settings.ts`.
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:20]:
```suggestion
locally, serving Settings Panel App, running tests, etc should be almost the same. Both types of apps have a controller for viewer part -`viewer.controller.ts`, entry points for worker bundles: `viewer.app.ts`, `editor.app.ts`, Settings panel, rendered via React  - `Settings.ts`.
```

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        │
+        └───ProductPage
+            .component.json
+            editor.controller.ts
+            viewer.controller.ts
+```
+
+### Components
+Each application consists of components. All components' files are grouped by directories located in `components` directory.
+For platform apps, main parts of the component are `editor.controller.ts` and `viewer.controller.ts`.
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:47]:
```suggestion
For platform apps, the main parts of the component are `editor.controller.ts` and `viewer.controller.ts`.
```

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    }
+}
+
+export default widgetScriptBuilder()
+  .withCreateMethod(createViewerController)
+  .build();
+```
+
+#### `editor.controller.ts`
+Controller for editor part. It's going to be collected and processed in `editorScript` under the hood.
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:70]:
```suggestion
A Controller for the editor part. Under the hood, it's going to be collected and processed in the `editorScript`.
```

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+}
+```
+
+
+### App Entry Points (`viewer.app.ts`, `editor.app.ts`)
+Both `viewerScript` (bundle that maps user's `viewer.controller`s with controller configs inside `createControllers` function) and `editorScript` (bundle that collects all `editor.controller`s in a single place) are generated by yoshi automatically.
+But you can still override values which are supported by the platform for these kinds of files.
+
+#### `viewer.app.ts`
+Here we'll map lifecycle methods by platform plus add useful helpers by the flow. Note, you can't override `createControllers`.
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:97]:
```suggestion
Here we'll map lifecycle methods provided by platform and add useful helpers provided by the flow. Note that you can't override `createControllers` function.
```

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+    return { someInitialData }; // Will be available in `viewer.controller`
+};
+
+export const initAppForPage = () => {
+    console.log('App inited!');
+};
+```
+
+#### `editor.app.ts`
+Entry-point for `editor` bundle. For more info read [editor app docs](https://bo.wix.com/wix-docs/client/editor-platform/platform-articles/application-structure-and-lifecycle#editor-platform_platform-articles_application-structure-and-lifecycle_application-life-cycle-and-structure)
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:118]:
```suggestion
An Entry-point for the `editor` bundle. For more info read [editor app docs](https://bo.wix.com/wix-docs/client/editor-platform/platform-articles/application-structure-and-lifecycle#editor-platform_platform-articles_application-structure-and-lifecycle_application-life-cycle-and-structure)
```

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+You can use `editorScriptBuilder` to build editor script by configuring it using your manifest, app's event handler and registering editor.controller.
+
+In the future, we'll reduce boilerplate and will register widget under the hood as we do for `viewerScript`.
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:121]:
```suggestion
In the future, we'll register the widget under the hood as we do for `viewerScript`.
```

So if I understand what you say, this file will be a bit redundant?

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+export default editorScriptBuilder()
+  .withEditorReady(editorReady)
+  .withAppManifest(appManifest)
+  .withEventHandler(eventHandler)
+  .withWidget(productWidget)
+  .build();
+```
+
+### App configuration files (`.application.json`, `.component.json`)
+As you noticed, we are providing some 'magic' under the hood to reduce boilerplate in many parts of the app. We are going to improve it in the future, so one day we'll simplify `editor.app.ts` as much as we can.
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:135]:
```suggestion
As you noticed, we are using "magic" to reduce boilerplate in many parts of the app. We are going to improve it in the future, so in the future we're going to simplify `editor.app.ts` as well.
```

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+To bring it to life we need some pre-configuration from the user.
+
+#### `.application.json`
+This is a configuration JSON object located in the project's root. Here we should specify `"appDefinitionId"` from the Dev Center. Yoshi uses it to override URLs during the local experience to point viewerScript and editorScript bundles to the localhost served by flow.
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:139]:
```suggestion
This is a configuration JSON object located in the project's root. Here we should specify `"appDefinitionId"` from the Dev Center. Yoshi uses it to override URLs during the local development to point viewerScript and editorScript bundles to the local code served by flow.
```

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+#### `.component.json`
+Each component should contain a single configuration file in the component's directory root. We support only `"id"` field for it with Component's ID from the Dev Center. It's needed for `createControllers` logic and URL overrides for the local experience.
```
Comment (ranyitz) [inline packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:141]:
```suggestion
Each component should contain a single configuration file in the component's directory root. We support only `"id"` field for it with Component's ID from the Dev Center. It's needed in order to match between each controller config to its matching controller logic and for URL overrides for local development.
```

File: packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md (hunk 9 above)
Comment (yavorsky) [reply packages/yoshi-flow-editor/docs/PLATFORM_APPS_SUPPORT.md:121]: No, you still could have it for platform lifecycle methods, but you don't have to register widget here in future.

**Pull request merged**